File: projects/GPL-FH-CSharp/features/WeightedWithNeighbors/Vertex.cs
//layer Weighted;

// of Graph
 
// *************************************************************************
 
public class Vertex {//refines
    public void AddWeight( Vertex end, int theWeight ) 
    {
        Neighbor the_neighbor =
                (Neighbor)(end.adjacentNeighbors[end.adjacentNeighbors.Count - 1]);
        end.adjacentNeighbors.RemoveAt(end.adjacentNeighbors.Count - 1);

        the_neighbor.weight = theWeight;
        ( end.adjacentNeighbors ).Add( the_neighbor );
    }
    
    public void adjustAdorns( Neighbor sourceNeighbor )
     {
        Neighbor tarGetNeighbor =
                (Neighbor)adjacentNeighbors[adjacentNeighbors.Count - 1];
        tarGetNeighbor.weight = sourceNeighbor.weight;
        original( sourceNeighbor );
    }
    
    public void display()
    {
        original();
    }

}

File: plugins/de.ovgu.featureide.examples/featureide_examples/GPL-FH-Java/features/WeightedWithNeighbors/GPL/Vertex.java
package GPL;

public class Vertex {
    public void addWeight( Vertex end, int theWeight ) 
    {
        Neighbor the_neighbor = 
                ( Neighbor ) ( end.adjacentNeighbors ).removeLast();
        the_neighbor.weight = theWeight;
        ( end.adjacentNeighbors ).add( the_neighbor );
    }
    
    public void adjustAdorns( Neighbor sourceNeighbor )
     {
        Neighbor targetNeighbor = 
                ( Neighbor )adjacentNeighbors.getLast();
        targetNeighbor.weight = sourceNeighbor.weight;
        original( sourceNeighbor );
    }
    
    public void display()
    {
        original();
    }

}
